pkg/controller/egress/validate.go

```diff
@@ -65,6 +65,10 @@ func (c *EgressController) ValidateEgress(review *admv1.AdmissionReview) *admv1.
 				return false, fmt.Sprintf("Burst %s in Egress %s is invalid: %v", newEgress.Spec.Bandwidth.Burst, newEgress.Name, err)
 			}
 		}
+		// Don't Allow if the FailurePolicy is updated.
+		if oldEgress.Spec.FailurePolicy != "" && newEgress.Spec.FailurePolicy != oldEgress.Spec.FailurePolicy {
```

Comment:
why it can't be updated?

Reply:
It can't be updated for now because once we have installed the flows to drop the traffic and then someone updates to NodeSNAT then we need to first uninstall the flows for drop, so we have decided to make it mutable as part of a separate task as an enhancement.
whats your take on this @tnqn

Reply:
I'm not sure if I get the challenge here. If the Egress doesn't "fail", updating `failurePolicy` should cause no change to any flows. If the Egress "fails", updating `failurePolicy` from one option to another doesn't sound more complex than other updates made to Egress like Egress IP change, Pod change.

We already have some technical debt to address to support reasonable updates to existing APIs. Unless there are significant challenges in implementing them, I’d prefer not to reject such updates, otherwise the user experience is not good when we suggest users to update the policy to fix some problems.

Reply:
done, the field is mutable now.